File: app/src/main/java/me/tagavari/airmessage/helper/CalendarHelper.java
package me.tagavari.airmessage.helper;

import java.util.Calendar;

public class CalendarHelper {
	/**
	 * Compares two calendars by their date (ignores time)
	 * @param cal1 The first calendar to compare
	 * @param cal2 The second calendar to compare
	 * @return the value {@code 0} if {@code cal1 == cal2};
	 *         a value less than {@code 0} if {@code cal1 < cal2}; and
	 *         a value greater than {@code 0} if {@code cal1 > cal2}
	 */
	public static int compareCalendarDates(Calendar cal1, Calendar cal2) {
		if(cal1.get(Calendar.ERA) < cal2.get(Calendar.ERA)) return -1;
		if(cal1.get(Calendar.ERA) > cal2.get(Calendar.ERA)) return 1;
		if(cal1.get(Calendar.YEAR) < cal2.get(Calendar.YEAR)) return -1;
		if(cal1.get(Calendar.YEAR) > cal2.get(Calendar.YEAR)) return 1;
		return Integer.compare(cal1.get(Calendar.DAY_OF_YEAR), cal2.get(Calendar.DAY_OF_YEAR));
	}
}

File: common/me/tagavari/airmessage/common/helper/CalendarHelper.kt
package me.tagavari.airmessage.common.helper

import java.util.*

object CalendarHelper {
	/**
	 * Compares two calendars by their date (ignores time)
	 * @param cal1 The first calendar to compare
	 * @param cal2 The second calendar to compare
	 * @return the value `0` if `cal1 == cal2`;
	 * a value less than `0` if `cal1 < cal2`; and
	 * a value greater than `0` if `cal1 > cal2`
	 */
	@JvmStatic
	fun compareCalendarDates(cal1: Calendar, cal2: Calendar): Int {
		return when {
			cal1[Calendar.ERA] < cal2[Calendar.ERA] -> -1
			cal1[Calendar.ERA] > cal2[Calendar.ERA] -> 1
			cal1[Calendar.YEAR] < cal2[Calendar.YEAR] -> -1
			cal1[Calendar.YEAR] > cal2[Calendar.YEAR] -> 1
			else -> cal1[Calendar.DAY_OF_YEAR].compareTo(cal2[Calendar.DAY_OF_YEAR])
		}
	}
}